util/set.go

```diff
@@ -0,0 +1,33 @@
+package util
+
+import (
+	"reflect"
+)
+
+type Set map[interface{}]bool
+
+func NewSet() *Set {
+	return new(Set)
+}
+
+func (set Set) Append(element interface{}) Set {
```

Comment:
1. Okay will start to create the pointer hell :P

2. I've first done that way first, I wanted a solution to this based on method overloading, but since I need the same type for both, makes sense to break in two.